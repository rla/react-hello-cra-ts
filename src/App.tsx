import { BrowserRouter, Route, Switch } from 'react-router-dom';
import { CountPage } from './components/CountPage';
import { ProfilePage } from './components/ProfilePage';
import logo from './logo-turnit.svg';

export const App = () => {
  return (
    <div className="App">
      <img src={logo} alt="Logo" width="400" />
      <BrowserRouter>
          <Switch>
            <Route exact={true} path="/" component={CountPage} />
            <Route exact={true} path="/profile" component={ProfilePage} />
          </Switch>
        </BrowserRouter>
    </div>
  );
};
