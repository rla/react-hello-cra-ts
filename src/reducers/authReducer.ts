import { AuthAction } from '../actions/auth';

export type AuthState = {
  loggedIn: boolean;
};

export const initialAuthState: AuthState = {
  loggedIn: false
};

export const authReducer = (state = initialAuthState, action: AuthAction) => {
  switch (action.type) {
    default:
      return state
  }
}
