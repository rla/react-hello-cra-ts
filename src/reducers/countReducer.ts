import { CountAction, SET_COUNT, INCREASE_COUNT } from '../actions/count';

export type CountState = {
  count: number;
};

export const initialCountState: CountState = {
  count: 0
};

type DeepReadonly<T> = { readonly [K in keyof T]: DeepReadonly<T[K]> };

export const countReducer = (
    state: DeepReadonly<CountState> = initialCountState,
    action: CountAction
  ) => {
  switch (action.type) {
    case SET_COUNT:
      // state.count = 2;
      return { ...state, count: action.value };
    case INCREASE_COUNT:
      return { ...state, count: state.count + 1 };
    default:
      return state;
  }
}
