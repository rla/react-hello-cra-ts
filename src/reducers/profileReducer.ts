import { ProfileAction, SET_PROFILE_DATA } from '../actions/profile';
import { ProfileData } from '../ProfileData';

export type ProfileState = {
  data?: ProfileData;
};

export const initialProfileState: ProfileState = {};

export const profileReducer = (state = initialProfileState, action: ProfileAction) => {
  switch (action.type) {
    case SET_PROFILE_DATA:
      return { ...state, data: action.data };
    default:
      return state;
  }
};
