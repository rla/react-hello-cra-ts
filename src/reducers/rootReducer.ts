import { combineReducers } from 'redux'
import { authReducer, AuthState } from './authReducer';
import { countReducer, CountState } from './countReducer'
import { profileReducer, ProfileState } from './profileReducer';

export type RootState = {
  auth: AuthState;
  count: CountState;
  profile: ProfileState;
};

export const rootReducer = combineReducers<RootState>({
  auth: authReducer,
  count: countReducer,
  profile: profileReducer
});
