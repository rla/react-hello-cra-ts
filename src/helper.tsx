import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import { rootReducer } from './reducers/rootReducer';

export const snapshotTest = (Component: any) => {
  const store = createStore(rootReducer);
  expect((mount(
    <Provider store={store}>
      <Component />
    </Provider>
  ).html())).toMatchSnapshot();
};
