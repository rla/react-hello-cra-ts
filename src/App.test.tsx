import { render, screen } from '@testing-library/react';
import { App } from './App';
import { createStore } from 'redux';
import { rootReducer } from './reducers/rootReducer';
import { Provider } from 'react-redux';
import { mount } from "enzyme";
import { snapshotTest } from './helper';

test('renders learn react link', () => {
  const store = createStore(rootReducer);
  render(
    <Provider store={store}>
      <App  />
    </Provider>
  );
  const linkElement = screen.getByText(/Vajutatud: 0/);
  expect(linkElement).toBeInTheDocument();
});

test('click count button and correct display', () => {
  const store = createStore(rootReducer);
  render(
    <Provider store={store}>
      <App />
    </Provider>
  );
  const button = screen.getByTestId('plusone-button');
  button.click();
  const spanElement = screen.getByTestId('display-count');
  expect(spanElement).toHaveTextContent('Vajutatud: 1');
});

test('app snapshot is stable', () => {
  snapshotTest(App);
});
