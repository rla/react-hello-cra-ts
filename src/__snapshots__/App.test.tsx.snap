// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app snapshot is stable 1`] = `"<div class=\\"App\\"><img src=\\"logo-turnit.svg\\" alt=\\"Logo\\" width=\\"400\\"><div style=\\"padding: 100px;\\"><div><button data-testid=\\"plusone-button\\" type=\\"button\\">+1</button></div><div><span data-testid=\\"display-count\\">Vajutatud: 0</span></div></div></div>"`;
